### PR TITLE
PKPDX-15 PKPDX-25

### PR DESCRIPTION
**Pull request by wniestroj**
[screencast-nimbusweb_me-2024_10_01-18_08_08.webm](https://github.com/user-attachments/assets/900a8bc9-5e6c-4efe-9b7f-756067deadc0)

### CODE REVIEW

File: frontend-v2/src/features/data/MapHeaders.tsx [hunk 1]
```diff
@@ -39,55 +49,81 @@ const MapHeaders: FC<IMapHeaders> = ({
...
+      <TablePagination
+        rowsPerPageOptions={rowsPerPageOptions}
+        component="div"
+        count={data?.length || 0}
+        rowsPerPage={rowsPerPage}
+        page={page}
+        onPageChange={handlePageChange}
+        onRowsPerPageChange={handleChangeRowsPerPage}
+        sx={{ display: 'flex', justifyContent: 'center', flexShrink: '0'}}
+      />
```
Comment (eatyourgreens) [inline frontend-v2/src/features/data/MapHeaders.tsx:125]: This works but `TablePagination` should be placed inside a `TableFooter`, inside `TableContainer`. See the [MUI Table examples](https://mui.com/material-ui/react-table/).

**Review by eatyourgreens — APPROVED** [review]
Changes look good. Just one comment about placing the new pagination controls inside the data table.

**Comment by wniestroj** [comment]
@eatyourgreens Merge request has been updated with additional changes regarding the notifications and some styling changes. Please review again. thank you!

File: frontend-v2/src/features/data/LoadDataStepper.tsx [hunk 2]
```diff
@@ -151,6 +155,8 @@ const LoadDataStepper: FC<IStepper> = ({ csv = "", onCancel, onFinish }) => {
   ) {
     errors = [...errors, "Time unit is not defined."];
   }
+  const showData = state.data.length > 0 && state.fields.length > 0;
+  const notificationsCount = errors?.length + warnings?.length + 2;
```
Comment (eatyourgreens) [inline frontend-v2/src/features/data/LoadDataStepper.tsx:159]: Out of interest, why is the default notifications count set to 2?

File: frontend-v2/src/features/data/LoadDataStepper.tsx [hunk 3]
```diff
@@ -261,7 +267,7 @@ const LoadDataStepper: FC<IStepper> = ({ csv = "", onCancel, onFinish }) => {
         width: "100%",
       }}
     >
-      <Stepper nonLinear activeStep={stepState.activeStep} alternativeLabel>
+      <Stepper nonLinear activeStep={stepState.activeStep}>
```
Comment (eatyourgreens) [inline frontend-v2/src/features/data/LoadDataStepper.tsx:270]: I think I used`alternativeLabel` here to make the step labels clickable, but I don't know if that's still needed in MUI v6.

Comment (eatyourgreens) [reply frontend-v2/src/features/data/LoadDataStepper.tsx:270]: Looks like the step button still work, without `alternativeLabel`.

Comment (eatyourgreens) [reply frontend-v2/src/features/data/LoadDataStepper.tsx:270]: My mistake: I used `nonLinear` to make the steps clickable. Ignore my previous comments.

File: frontend-v2/src/features/data/LoadDataStepper.tsx (hunk 2 above)
Comment (eatyourgreens) [reply frontend-v2/src/features/data/LoadDataStepper.tsx:159]: Sounds good!

**Pull request merged**